tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c

```diff
@@ -19,8 +19,17 @@ BUILD_ASSERT(NUM_THREAD <= MAX_NUM_THREAD);
 #define BUSY_MS (SLICE_SIZE + 20)
 /* a half timeslice*/
 #define HALF_SLICE_SIZE (SLICE_SIZE >> 1)
-#define HALF_SLICE_SIZE_CYCLES \
-	((u64_t)(HALF_SLICE_SIZE) * sys_clock_hw_cycles_per_sec() / 1000)
+#define HALF_SLICE_SIZE_CYCLES                                                 \
+	((u64_t)(HALF_SLICE_SIZE)*sys_clock_hw_cycles_per_sec() / 1000)
+
+/* Task switch tolerance ... */
+#if CONFIG_SYS_CLOCK_TICKS_PER_SEC >= 1000
+/* ... will not take more than 1 ms. */
+#define TASK_SWITCH_TOLERANCE (1)
+#else
+/* ... 1ms is faster than a tick, loosen tolerance to 1 tick */
+#define TASK_SWITCH_TOLERANCE (1000 / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
```

Comment:
This is vulnerable to a repeated-truncation problem that's showed up multiple times in the past.  For example, if `CONFIG_SYS_CLOCK_TICKS_PER_SEC` is 256 the tolerance should be 3.90625 ms, but it'll be truncated to 3 ms.  In actual use durations of 256 ticks will appear to be either 3 or 4 ms long, averaging to the true value.

Below the tolerance is used to calculate both a lower bound and an upper bound, where it is again truncated.  The problem is that for the lower bound it's appropriate to subtract 3, but for the upper bound you must add 4.  If you only add 3 you get intermittent failures in the cases where the experienced duration maps to 4 ms.

This is exactly the sort of thing that #19591 will help with, by providing conversion functions that round in the direction necessary for the specific use, so in hopes that we get that unstuck and merged soon I'm ok with merging this as is, with a plan to replace the calculations with accurate ones once infrastructure is available to support that.

Reply:
Thank you @pabigot and @ioannisg.
I changed the title and description to reflect the new purpose of this PR.

Should we have this PR merged and then add a new PR for mathematical rounding or would something like `ceiling_fraction` macro do the job in a new force push? @pabigot

Reply:
Yes, I'd like this to go in as-is.  We're waiting on #19591 to provide `k_ticks_to_ms_ceil32()` API to help with the potential concern I identified.